contracts/smart-contract-wallet/BaseSmartAccount.sol

```diff
@@ -55,12 +55,13 @@ abstract contract BaseSmartAccount is IAccount, BaseSmartAccountErrors {
     /**
      * Validate user's signature and nonce.
      * subclass doesn't need to override this method. Instead, it should override the specific internal validation methods.
+     * @dev name is optimized for this method to be cheaper to be called
      */
-    function validateUserOp(
+    function validateUserOp_kab(
         UserOperation calldata userOp,
         bytes32 userOpHash,
         uint256 missingAccountFunds
-    ) external virtual override returns (uint256 validationData) {
+    ) public returns (uint256 validationData) {
```

Comment:
Entry point will always call validateUserOp on the account as per IAccount interface, so we should / need not keep this one.

Reply:
you mean there's no way anything will call `validateUserOp_kab` directly?

Reply:
yes, its very unlikely that any entity except entry point will call this method. and entry point will always call validateUserOp

Reply:
removed